src/main/java/com/dongsoop/dongsoop/member/service/MemberService.java

```diff
@@ -0,0 +1,90 @@
+package com.dongsoop.dongsoop.member.service;
+
+import com.dongsoop.dongsoop.exception.domain.member.*;
+import com.dongsoop.dongsoop.member.dto.SignupRequest;
+import com.dongsoop.dongsoop.member.entity.Member;
+import com.dongsoop.dongsoop.member.entity.Role;
+import com.dongsoop.dongsoop.member.repository.MemberRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.regex.Pattern;
+
+@Service
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+public class MemberService {
+
+    private final MemberRepository memberRepository;
+    private final PasswordEncoder passwordEncoder;
+
+    private static final String EMAIL_REGEX = "^[A-Za-z0-9+_.-]+@[A-Za-z0-9.-]+$";
+    private static final String PASSWORD_REGEX = "^(?=.*[A-Za-z])(?=.*\\d)(?=.*[@$!%*#?&])[A-Za-z\\d@$!%*#?&]{8,}$";
+    private static final String STUDENT_ID_REGEX = "^\\d{8}$";
+
+    @Transactional
+    public void signup(SignupRequest request) {
+        checkEmailDuplication(request.getEmail());
+        validateSignupRequest(request);
+
+        Member member = Member.builder()
+                .email(request.getEmail())
+                .password(passwordEncoder.encode(request.getPassword()))
+                .nickname(request.getNickname())
+                .studentId(request.getStudentId())
+                .department(request.getDepartment())
+                .role(Role.USER)
+                .build();
+
+        memberRepository.save(member);
+    }
+
+    public Member findById(Long id) {
+        return memberRepository.findById(id)
+                .orElseThrow(MemberNotFoundException::new);
+    }
+
+    public Member findByEmail(String email) {
+        return memberRepository.findByEmail(email)
+                .orElseThrow(MemberNotFoundException::new);
+    }
+
+    private void checkEmailDuplication(String email) {
+        memberRepository.findByEmail(email)
+                .ifPresent(m -> {
```

Comment:
한 단계 더 나누어 코드를 작성하면 가독성이 향상될 것 같습니다.
또한 익명함수 사용 시 파라미터 영역에 소괄호를 사용하여 명확하게 영역을 지정해주는 것이 좋아보입니다.
```suggestion
        Optional<Member> optionalMember = memberRepository.findByEmail(email);
        optionalMember.ifPresent((m) -> {
            throw new EmailDuplicatiedException()
        });
```

Reply:
수정완료